### ISSUE TITLE
fail2ban unbans/rebans on jail reload command

### ISSUE THREAD

**Issue by dsmithdewarcom** [fail2ban/fail2ban#2132]
- Fail2Ban version (including any possible distribution suffixes): 0.8.11-1
- OS, including release name/version: Ubuntu 14.04
- [x ] Fail2Ban installed via OS/distribution mechanisms
- [x ] You have not applied any additional foreign patches to the codebase
- [ x] Some customizations were done to the configuration (provide details below is so)

### The issue:
When reloading a jail after modifying its filter, f2b removes the ip's from iptables previously banned via this jail, then adds them back.

(not sure this is an issue per se, but it sure would be nice if there were a reload flag to prevent this, e.g. "fail2ban-client -k reload myjail";  "k" = keep the banz (or whatever)

#### Steps to reproduce
- modify jail filter
- sudo fail2ban-client reload [myjail]

#### Expected behavior* (maybe this should be "Preferred behavior"?)
f2b reloads jail without removing rules from iptables, but adds new ones per jail modifications.

#### Observed behavior
f2b strips ip's from iptables previously banned via this jail, then (eventually) rebans them

#### Any customizations done to /etc/fail2ban/ configuration
jail.local:
[myjail]
enabled  = true
port     = http,https
filter   = myfilter
logpath  = /var/log/apache2/access.log
maxretry = 3
findtime = 259200
bantime  = 1814400

myfilter:
[INCLUDES]
# overwrite with apache-common.local if _apache_error_client is incorrect.
before = apache-common.conf
[Definition]
failregex = ^<HOST> - - .*("GET \/phpmyadmin|"GET \/phpMyAdmin|"GET \/pma\/|"GET \/myadmin|"GET \/admin|"GET \/mysql|"GET \/lfm.php|"GET \/sftp|"GET \/file|"GET \/site|"GET \/winscp|"POST \/ HTTP).*$
ignoreregex = ''

#### Relevant parts of /var/log/fail2ban.log file:

2018-05-18 11:39:52,490 fail2ban.actions: WARNING [myjail] Unban xxx.xxx.xxx.xxx
[etc]
2018-05-18 11:39:52,490 fail2ban.actions: WARNING [myjail] Ban xxx.xxx.xxx.xxx
[etc]

**Comment by sebres** [comment]
Reload in version previous 0.10 works like restart.
This handling changed in 0.10, and reload does not reban (if not expected, e. g. if action was not affected).

**Comment by dsmithdewarcom** [comment]
I found a 0.10x backport for trusty (14.04) here:
https://launchpad.net/~jan-hoffmann/+archive/ubuntu/fail2ban

This preserved my incompatible, old version jail.local, and was generally broken out of the box, until I rebuilt jail.local from the new version jail.conf

**Comment by sebres** [comment]
> until I rebuilt jail.local from the new version jail.conf

[wiki :: Proper fail2ban configuration](https://github.com/fail2ban/fail2ban/wiki/Proper-fail2ban-configuration)

Shortly: to hold it maintainable, don't copy jail.conf into jail.local - only your custom changes should be placed there.

**Comment by dsmithdewarcom** [comment]
thanks, that makes it much easier to parse. I am still getting "No file(s) found for glob /var/log/nginx/*access.log" on fail2ban reloads, even though I don't have nginx installed and the relevant jails aren't enabled.

**Comment by sebres** [comment]
Hmm... Although this may be just a warning (by multiple paths), if so you can ignore this.

If you see any logpath in the output of:
```bash
fail2ban-client -d | grep "/nginx/"
```
you should search for some active config (grep, etc), that is affected here.

**Comment by dsmithdewarcom** [comment]
Yeah, they're in paths-common.conf (possibly elsewhere), I tried deleting those entries earlier but that just led to a reload fail.

**Comment by dsmithdewarcom** [comment]
"nginx" does not appear in output of fail2ban-client -d

**Comment by sebres** [comment]
If you don't see "Have not found any log file for ...", then it is just a warning (you've some jail which uses multiple logpaths), so ignore it.

But indeed, we should change the logging level (error to notice/info). 
Thx.

**Comment by sebres** [comment]
Fixed in a3cb1dbd4d93079af44c5186ac8cf6eb1a886421.
As long as not released, just ignore this output.
Thx again.
